syntax = "proto2";

package object_detection.protos;

import "layer_models/protos/eval.proto";
import "layer_models/protos/graph_rewriter.proto";
import "layer_models/protos/input_reader.proto";
import "layer_models/protos/model.proto";
import "layer_models/protos/train.proto";

// Convenience message for configuring a training and eval pipeline. Allows all
// of the pipeline parameters to be configured from one file.
message TrainEvalPipelineConfig {
  optional Model model = 1;
  optional TrainConfig train_config = 2;
  optional InputReader train_input_reader = 3;
  optional EvalConfig eval_config = 4;
  optional InputReader eval_input_reader = 5;
  optional GraphRewriter graph_rewriter = 6;
  extensions 1000 to max;
}
